### ISSUE TITLE
fix: Issue with Black Screen on iOS After Hotfix Update Using Different Shorebird Versions

### ISSUE THREAD

**Issue by JedTsai33** [shorebirdtech/shorebird#2667]
App ID: 0201f7bd-11de-4ad3-af2b-d8b317cb1c33

**Description**

We are using GitHub to package the `.ipa` file and installing Shorebird via` shorebirdtech/setup-shorebird@v1.`

Previously, we released a version using Shorebird **v1.4.1**. Today, we attempted to publish a patch update using Shorebird **v1.4.15**. However, after the hotfix was applied, opening the app resulted in a black screen.

Through further testing, we found that:

If both the release and the patch are built with Shorebird **v1.4.15**, the hotfix works correctly without any issues.

Questions:
1. Why does updating to a newer version of Shorebird cause the iOS app to display a black screen after a hotfix?
2. Is it possible to let users control the Shorebird version during installation to avoid similar issues in the future?

**Comment by eseidel** [comment]
I don't believe this has anything to do with the Shorebird version.  (It's possible, but unlikely.) Shorebird always uses the *exact* Flutter matching the release when building a patch, regardless of Shorebird version.

But it does look like 7.100.5+710005 patch 5, had native changes, which easily could explain a black screen:

```
  "has_asset_changes": true,
  "has_native_changes": true,
  "release_platform": "ios",
  "used_ignore_asset_changes_flag": true,
  "used_ignore_native_changes_flag": true
```

I'd strongly recommend not ignoring native changes.
https://docs.shorebird.dev/troubleshooting/#i-see-a-your-app-contains-native-changes-warning-when-running-shorebird-patch-even-though-i-havent-changed-swiftobjective-ckotlinjava-code

The way that Shorebird works is that it replaces all the Dart code in your app.  It does not change any of the "native" code (Obj-C/Swift/Kotlin/Dart/C++, etc.), so if you *have* changed native code and patch, you will end up running *new* Dart code with *old* "native" code, and thus can end up in a confused state.  `shorebird` warns if this happens, but it looks like when these patches were made those warnings were overridden.

I've filed https://github.com/shorebirdtech/shorebird/issues/2669 for displaying a warning about this in the console.

**Comment by eseidel** [comment]
Can you share the command you used to build the patch?

**Comment by JedTsai33** [comment]
**We used the following patch command:**
`shorebird patch ios --release-version=7.2.1+7201 --flavor $FLAVOR --no-codesign --allow-native-diffs --allow-asset-diffs`

After reviewing the backend logs, we observed that a patch released on 2024/11/25 was functioning correctly. At that time, we were using Shorebird v1.4.13, which indicates that versions prior to v1.4.13 had no issues.

However, when using Shorebird v1.4.15 to release a patch on 2024/12/9, the app displayed a black screen on iOS.

Through further cross-comparison testing, we found:
- When the release is built with v1.4.15 and the patch is also created with v1.4.15, the patch includes all the necessary modifications and updates normally without issues.

Based on these findings, we suspect that Shorebird v1.4.15 is not backward compatible with iOS releases built using earlier versions of Shorebird.
